decidim/decidim
# Improve navigation and filters in proposals

#### :tophat: What? Why?

This PR:
 * Adds a system for admins to store in session their filters choices which are recovered whenever they enter in the admin index of a resource with filters without filters options in the query string part of the URL. If there are filters specified the admin is redirected to the path which includes the filters in the query. This redirection is made calling a `check_admin_session_filters` method defined in the filterable concern which has been added only to proposals index, but it can be used in other indexes with filters of the admin panel to keep the filters choices
 * Adds an option to the filters query string which allows admins to reset their filter configuration on each specific index with filters
 * Provides in the tags of filters a specific link to reset all the filters and return to the index without filters options in the query string
 * When an admin opens the show view of a proposal from a filtered selection on the index of proposals the filter is stored and used to generate the links of previous and next proposal on the filtered list if any. If the admin returns to the index the filter is recovered (
 * This previous and next adjacent items feature can be used in other parts of the application combined with the storage of the filters state. The only requirement is that the controller uses the filterable concern
 
#### :pushpin: Related Issues
*Link your PR to an issue*
- Fixes  #12908

#### Testing
As admin:

* Visit the index of a proposals component: https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/4/manage/proposals
![image](https://github.com/decidim/decidim/assets/446459/c9e90d08-fafc-45c8-b866-c7f81f425240)

* Check some filters: The URL should change accordingly including the selected filters. If there are more than one checked filter an option to remove all filters appears: https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/4/manage/proposals?q%5Bis_emendation_true%5D=false&q%5Bwith_any_state%5D=state_not_published
![image](https://github.com/decidim/decidim/assets/446459/bcf5cfe3-87f1-46ea-b6d5-421e186d1462)

* Visit an item of the filtered list. A previous and / or next links should appear including items of the filtered list. https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/4/manage/proposals/7
![image](https://github.com/decidim/decidim/assets/446459/173b203d-a993-448b-89b3-bc07fc63edb5)

* Go back to the proposals index: https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/4/manage/proposals. The previously selected filters should appear and the application should redirect to the path including the filters
![image](https://github.com/decidim/decidim/assets/446459/f5a3c83e-8ab2-4761-b0d3-ecdccb8a2ed8)


:hearts: Thank you!
